src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php

```diff
@@ -119,4 +125,32 @@ private function computeSignatureHash(UserInterface $user, int $expires): string
 
         return base64_encode(hash_hmac('sha256', implode(':', $signatureFields), $this->secret));
     }
+
+    private function encrypt(string $message): string
+    {
+        if (class_exists(Crypto::class)) {
+            return Crypto::encrypt(new HiddenString($message), $this->createKey());
+        }
+
+        return $message;
+    }
+
+    private function decrypt(string $message): string
+    {
+        try {
+            return Crypto::decrypt($message, $this->createKey());
+        } catch (InvalidMessage $e) {
+            // TODO Catch other exceptions?
+            return $message;
+        }
+    }
+
+    private function createKey(): EncryptionKey
+    {
+        // TODO make sure we dont use a static salt.
+        // TODO store salt some how between requests
+        $salt = "\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f";
+
+        return KeyFactory::deriveEncryptionKey(new HiddenString($this->secret), $salt, KeyFactory::MODERATE);
```

Comment:
I need help to figure this out. Using a static salt is not a good idea. 

The salt also needs to be kept secret. We must store the salt somehow.

Reply:
As a salt is merely used to prevent that a same value would always produce the same hash/key. Would it make sense to embed with the message? Encryption uses an Initialization Vector for a similar purpose, and that is part of the message as well.

Maybe you can ping the creator behind Halite for some advice?